packages/plugin-hardhat/test/transparent-admin-validation.js

```diff
@@ -12,10 +13,16 @@ test.before(async t => {
 test('admin validation', async t => {
   const { Greeter, GreeterV2 } = t.context;
   const greeter = await upgrades.deployProxy(Greeter, ['Hola admin!'], { kind: 'transparent' });
-  await upgrades.admin.changeProxyAdmin(await greeter.getAddress(), NEW_ADMIN);
-  await t.throwsAsync(
-    () => upgrades.upgradeProxy(greeter, GreeterV2),
-    undefined,
-    'Proxy admin is not the one registered in the network manifest',
-  );
+
+  const [, signer] = await ethers.getSigners();
+  const AdminFactory = await getProxyAdminFactory(hre, signer);
+  const deployedAdmin = await AdminFactory.deploy();
+  const deployedAdminAddress = await deployedAdmin.getAddress();
+  await upgrades.admin.changeProxyAdmin(await greeter.getAddress(), deployedAdminAddress);
+
+  // Old admin signer cannot upgrade this
+  await t.throwsAsync(() => upgrades.upgradeProxy(greeter, GreeterV2));
+
+  const GreeterV3 = Greeter.connect(signer);
+  await upgrades.upgradeProxy(greeter, GreeterV3);
```

Comment:
Can you add a test for running `changeProxyAdmin` again?  For example, add this here:

```
  // New admin signer can change the admin again
  const deployedAdmin2 = await AdminFactory.deploy();
  const deployedAdminAddress2 = await deployedAdmin2.getAddress();
  await upgrades.admin.changeProxyAdmin(await greeter.getAddress(), deployedAdminAddress2, signer);
  ```